### ISSUE TITLE
disallow unnecessary methods

### ISSUE THREAD

**Issue by chdsbd** [sbdchd/flake8-pie#65]
```python
class MyClass:
    name: str
    def update(self, name: str) -> None:
        self.name = name
    def some_func(self, count: int) -> str:
        return f"foo:{count}"
```

Should be

```python

def some_func(count: int) -> str:
    return f"foo:{count}"

class MyClass:
    name: str
    def update(self, name: str) -> None:
        self.name = name
```

**Comment by sbdchd** [comment]
Curious about false positives. If we had a function that takes a `Protocol` as a param and this `class` was trying to satisfy that `Protocol` with the `some_func` method, appeasing the lint would result in a breakage.

**Comment by chdsbd** [comment]
I was thinking about the protocol issue too. hmm.
